docs/products/kafka/concepts/upgrade-procedure.rst

```diff
@@ -0,0 +1,253 @@
+Apache Kafka® upgrade procedure
+===============================
+
+One of the benefits of using a managed service like Aiven for Apache Kafka® is the automated upgrade procedure. 
+
+The upgrade procedure is executed during:
+    
+* maintenance updates
+* plan changes
+* cloud region migrations
+* manual node replacements performed by an Aiven operator
+    
+All the above operations involve creating new broker nodes to replace existing ones.
+
+Upgrade procedure steps
+---------------------------
+
+The following set of steps are executed during an upgrade procedure, the example shows a possible scenario for a 3-node Apache Kafka service.
+
+.. mermaid::
+
+    flowchart TD;
+        subgraph KafkaCluster
+            subgraph Node3
+                PartitionB2([Partition B])
+                PartitionC2([Partition C])
+            end
+            subgraph Node2
+                PartitionA2([Partition A])
+                PartitionC1([Partition C])
+            end
+            subgraph Node1
+                PartitionA1([Partition A])
+                PartitionB1([Partition B])
+            end
+        end
+        
+    
+
+1. New Apache Kafka® nodes are started alongside the existing nodes
+
+.. mermaid::
+
+    flowchart TD;
+
+        subgraph NewNode3
+            PartitionNew31([ ])
+            PartitionNew32([ ])
+        end
+        subgraph NewNode2
+            PartitionNew21([ ])
+            PartitionNew22([ ])
+        end
+        subgraph NewNode1
+            PartitionNew11([ ])
+            PartitionNew12([ ])
+        end
+        subgraph KafkaCluster
+            subgraph Node3
+                PartitionB2([Partition B])
+                PartitionC2([Partition C])
+            end
+            subgraph Node2
+                PartitionA2([Partition A])
+                PartitionC1([Partition C])
+            end
+            subgraph Node1
+                PartitionA1([Partition A])
+                PartitionB1([Partition B])
+            end
+        end
+        
+
+2. Once the new nodes are running, they join the Apache Kafka cluster
```

Comment:
It's probably the behaviour of mermaid, but because the new nodes are visualised with different width/hights/margins, it kinda doesn't look too nice